providers/gce/gce_instances.go

```diff
@@ -551,19 +551,29 @@ func (g *Cloud) AliasRangesByProviderID(providerID string) (cidrs []string, err
 		for _, r := range networkInterface.AliasIpRanges {
 			cidrs = append(cidrs, r.IpCidrRange)
 		}
-		ipv6Addr := getIPV6AddressFromInterface(networkInterface)
-		if ipv6Addr != "" {
-			// The podCIDR range is the first /112 subrange from the /96 assigned to
-			// the node
-			ipv6PodCIDR := fmt.Sprintf("%s/112", ipv6Addr)
-			cidrs = append(cidrs, ipv6PodCIDR)
-		} else {
-			klog.Infof("No IPv6 addresses found for %s", providerID)
+		ipv6Addr := g.GetIPV6Address(networkInterface)
+		if ipv6Addr != nil {
+			cidrs = append(cidrs, ipv6Addr.String())
 		}
 	}
 	return
 }
 
+// GetIPV6Address fetches the IPv6 addressses associated with a network interface.
+func (g *Cloud) GetIPV6Address(networkInterface *compute.NetworkInterface) *net.IPNet {
+	ipv6Addr := getIPV6AddressFromInterface(networkInterface)
+	if ipv6Addr == "" {
+		return nil
+	}
+	addr := net.ParseIP(ipv6Addr)
+	return &net.IPNet{
+		IP: addr,
+		// The podCIDR range is the first /112 subrange from the /96 assigned to
+		// the node
+		Mask: net.CIDRMask(112, 128),
+	}
```

Comment:
https://go.dev/play/p/FbQ4lfFzbah

```
	addr := net.ParseIP(ipv6Addr)
	mask := net.CIDRMask(112, 128)
	return &net.IPNet{IP: addr.Mask(mask), Mask: mask}
```

Reply:
I'm trying to refactor following the existing implementation here - https://github.com/kubernetes/cloud-provider-gcp/blob/master/providers/gce/gce_instances.go#L554-L562

Reply:
Agree that we don't actually need the exact IP in the range, not sure why they were doing it in the first place.